File: app/src/main/java/social/entourage/android/guide/GuideMapComponent.java
package social.entourage.android.guide;

import dagger.Component;
import social.entourage.android.ActivityScope;
import social.entourage.android.EntourageComponent;

/**
 * Component linked to GuideMapActivity lifecycle
 * Provide a GuideMapPresenter
 * @see GuideMapActivity
 * @see GuideMapPresenter
 */
@ActivityScope
@Component(
        dependencies = EntourageComponent.class,
        modules = GuideMapModule.class
)
@SuppressWarnings("unused")
public interface GuideMapComponent {
    void inject(GuideMapActivity activity);

    GuideMapPresenter getGuideMapPresenter();
}

File: app/src/main/java/social/entourage/android/guide/GuideMapComponent.kt
package social.entourage.android.guide

import dagger.Component
import social.entourage.android.base.ActivityScope
import social.entourage.android.EntourageComponent

/**
 * Component linked to GuideMapFragment lifecycle
 * Provide a GuideMapPresenter
 * @see GuideMapFragment
 *
 * @see GuideMapPresenter
 */
@ActivityScope
@Component(dependencies = [EntourageComponent::class], modules = [GuideMapModule::class])
interface GuideMapComponent {
    fun inject(fragment: GuideMapFragment?)
    val guideMapPresenter: GuideMapPresenter?
}